apps/www/_blog/2022-06-28-partial-postgresql-data-dumps-with-rls.mdx

```diff
@@ -0,0 +1,178 @@
+
+```sql hideCopy
+-- Turn on Row Level Security
+alter table profiles
+  enable row level security;
+
+-- Only dump data for internal team members 1, 2, 3
+create policy "Data dump rule" on profiles
+  for select
+  to exporter
+  using (
+    id in (1, 2, 3)
+  );
+```
+
+### Step 3: Export the data
+
+Now we can use `pg_dump` to get only the data that we need.
+
+Run the dump with the `exporter` user that we created above and use the `--enable-row-security` flag to ensure that the dump succeeds.
+
+```sh hideCopy
+
+# Dump all the data into a "seed.sql" file
+# which we can use to restore our local databases.
+pg_dump \
+-h db.host.supabase.co \
+-U exporter \
+-d postgres \
+-n public \
+--data-only \
+--enable-row-security \
+--table=profiles \
+> seed.sql
+```
+
+And that's it. You can follow this same pattern for any tables that you want to dump.
+
+## Data access patterns
+
+RLS is a bit like appending a “where” clause to a `select`, so you can create all sorts of data access patterns. Let's see a few more which are useful for extracting seed data.
+
+### Using email rules
+
+Instead of using hardcoded numbers in our RLS policies, we could use email extensions to determine the users who we want to export:
+
+```sql hideCopy
+-- Only dump data for supabase employees
+create policy "Data dump rule" on profiles
+  for select
+  to exporter
+  using (
+    substring(email from '@(.*)$') = 'supabase.com'
+  );
+
+```
+
+### Only recent data
+
+If we have a table with a lot of data, like an `analytics` table, we might only care about the last 2 months of data.
+
+```sql hideCopy
+-- A fake analytics table where we store actions a user takes
+create table analytics (
+  id serial primary key,
+  ts timestamptz default now(),
+  profile_id references profiles,
+  event text
+);
+alter table profiles
+  enable row level security;
+
+-- Here is an "age" rule so that we only dump the most recent analytics
+create policy "Data dump rule" on logs
+  for select
+  to exporter
+  using (
+    profile_id in (1, 2, 3) and
+    ts > now() - interval '2 MONTHS' -- here's the magic
+  );
+```
+
+### Using flags
+
+If you don't mind having some additional columns in you database, you can add flags to each row to determine whether it's safe to export.
+
+```sql hideCopy
+create table profiles (
+  id serial primary key,
+  name text,
+  email text,
+  is_exportable boolean -- make this "TRUE" if you want to allow access
+);
+alter table profiles
+  enable row level security;
+
+-- Only dump data for internal team members 1, 2, 3
+create policy "Data dump rule" on profiles
+  for select
+  to exporter
+  using ( is_exportable = true );
+
+```
+
+## Conclusion
+
+Using `seed` data isn't the only way to run development environments. It's also possible to run fully-masked copies of your database using tools like
+[Snaplet](https://docs.snaplet.dev/tutorials/supabase-clone-environments).
```

Comment:
This link is not working for me: https://docs.snaplet.dev/tutorials/supabase-clone-environments

Reply:
Unfortunately it doesn't:

![Screenshot from 2022-06-30 16-04-37](https://user-images.githubusercontent.com/1829294/176778109-fb40ea33-a49e-40c6-8b2a-4648d3ef3d07.png)

Reply:
I'll ping the snaplet team